### ISSUE TITLE
 The receiving value not properly rounded

### ISSUE THREAD

**Issue by VolodLytvynenko** [status-im/status-react#21188]
reason of rounding https://github.com/status-im/status-desktop/issues/16267
follow up for https://github.com/status-im/status-mobile/pull/21134
Perhaps it would be good to fix it together with the related issue https://github.com/status-im/status-mobile/issues/21187

### Steps:
1. Go to the swap page and select ETH as for send and DAI as for receiving
2. Enter a minimum value, such as 0.0000000001 ETH.

###Actual Result:
The receiving value is displayed with all decimals in full, overflowing from the receiving section
![image](https://github.com/user-attachments/assets/b757ce10-c95b-46da-bfe3-4f3b92936d29)

### Expected Result:
Potential solutions:
- Crypto values should be rounded so that the final decimal place always represents a value between $0.01 and $0.10 USD, ensuring that small values are displayed appropriately.
OR
- The token value should be rounded to the "maximum number of decimal places" as specified in https://github.com/status-im/status-mobile/issues/18244

Examples:
- If the user enters a fiat value resulting in a token value with more than the allowed decimal places (e.g., 0.37 USD which equals 0.0001001... ETH), and taps the "show in token" button, the value in the input field should become 0.0001 ETH.)
- When ETH is priced at $365, four decimal places should be used for ETH because 0.0001 ETH = $0.036 USD, which is between $0.01 and $0.10 USD

### ENV:
Nightly 6 Sep, 2024

**Comment by briansztamfater** [comment]
We have a design for long numbers, so I think rounding is not needed in this case, wdyt? https://www.figma.com/design/AD2JSKg0I8dZcylyiGa62O/Swap-for-Mobile?node-id=98-58893&t=adKG1SwEaJN1qSZQ-4 cc @VolodLytvynenko @alwx

**Comment by VolodLytvynenko** [comment]
> We have a design for long numbers, so I think rounding is not needed in this case, wdyt? https://www.figma.com/design/AD2JSKg0I8dZcylyiGa62O/Swap-for-Mobile?node-id=98-58893&t=adKG1SwEaJN1qSZQ-4 cc @VolodLytvynenko @alwx

Hey @briansztamfater, unfortunately, we can't work with such decimals. It seems that the swap provider doesn't accept these values. Reported it here https://github.com/status-im/status-desktop/issues/16267

**Comment by briansztamfater** [comment]
> > We have a design for long numbers, so I think rounding is not needed in this case, wdyt? https://www.figma.com/design/AD2JSKg0I8dZcylyiGa62O/Swap-for-Mobile?node-id=98-58893&t=adKG1SwEaJN1qSZQ-4 cc @VolodLytvynenko @alwx
> 
> Hey @briansztamfater, unfortunately, we can't work with such decimals. It seems that the swap provider doesn't accept these values. Reported it here [status-im/status-desktop#16267](https://github.com/status-im/status-desktop/issues/16267)

Hey @VolodLytvynenko, I am sorry but now I am bit confused and I think I don't fully understand the bug behind this issue.

Looking at the description and screenshots of the issue, `0.0001` is a valid amount because `USDC` token has 6 decimals. On the other side, the receiving amount is returned from the provider, so in the case of is a `DAI` value with many decimals. `DAI` token has 18 decimals, so it seems like a correct amount too.

In the description of this issue it says:
> The receiving value is displayed with all decimals in full, overflowing from the receiving section

In my previous comment I pointed to a design that we can implement to display long overflowing numbers, which should fix how we display the long numbers in the receiving input (or, at least, improve the UX for that edge case). Rounding decimals is not needed, and I think we shouldn't do it because it would confuse the user on what are the real values of the swap.

About the issue from status-desktop you are mentioning, in #21261, where we are implementing error handling, if an error comes from Paraswap, it is shown as "Paraswap error: _[error message]_". I don't think we can do anything else if the error comes from the provider.

So, to summarize, I am not sure if the issue is about how we display overflowing numbers in the receiving input, or how we display an error coming from Paraswap, or if there's other issue that needs to be fixed. Can you clarify that, please?

Thanks and sorry for the confusion :)

**Comment by VolodLytvynenko** [comment]
Hi @briansztamfater, yes, I agree with you. The only thing needed here is fixing the UI, without doing any rounding in the "receive" section. I'll close this issue. Sorry for the confusion, and thank you for pointing out the mistake!

**Comment by briansztamfater** [comment]
Thanks @VolodLytvynenko!
